satpy/readers/yaml_reader.py

```diff
@@ -870,7 +870,114 @@ def _load_area_def(dsid, file_handlers):
     return final_area.squeeze()
 
 
-class GEOSegmentYAMLReader(FileYAMLReader):
+def _set_orientation(dataset, upper_right_corner='native', **kwargs):
+    """Set the orientation of datasets.
+
+    Allows to flip geostationary datasets without having to resample.
+
+    Args:
+        datasets (iterable): Limit the flipping to this list of names (str), wavelengths (float), or
+                            DatasetID objects.
+                            By default all currently loaded datasets are flipped.
+        upper_right_corner (str): Direction of the upper right corner of the image.
+                                Possible options are 'NW', 'NE', 'SW', 'SE', or 'native'.
+                                The common upright image orientation corresponds to 'NW'.
+                                Defaults to 'native' (no flipping is applied).
+
+    """
+    def _change_area_extent_of_areadef(areadef, new_extent):
+        """Change the area extent of an area definition and regenerate it."""
+        # does this need to be moved somewhere else?
+        return AreaDefinition(
```

Comment:
Can you use `aredef.copy(extent=new_extent)` here?

Reply:
Ah yes indeed, nice solution! I got rid of that function alltogether.